java/core/src/test/java/sleeper/core/statestore/inmemory/StateStoreTestBuilder.java

```diff
@@ -59,7 +59,15 @@ public StateStoreTestBuilder singleFileInEachLeafPartitionWithRecords(long recor
                 .map(partition -> partitionSingleFile(partition, records)));
     }
 
-    public StateStoreTestBuilder partitionFileWithRecords(String partitionId, String filename, long records) {
+    public StateStoreTestBuilder partialFileWithRecords(String partitionId, String filename, long records) {
+        return addFile(FileInfo.partialFile()
+                .fileStatus(FileInfo.FileStatus.ACTIVE)
+                .partitionId(partitionId)
+                .filename(filename)
+                .numberOfRecords(records).build());
+    }
+
+    public StateStoreTestBuilder wholeFileWithRecords(String partitionId, String filename, long records) {
```

Comment:
I feel like it confuses things a bit by specifying "whole file" in every test that uses this. Could we add the parent file first using the old method signature, and then have a separate method to split it to the child partitions?

I added a utility for splitting a FileInfo in the branch this PR is based from.